### ISSUE TITLE
[Regression] Add-Type fails with CS1701 when referencing Newtonsoft.Json on daily build of PS Core

### ISSUE THREAD

**Issue by kborowinski** [PowerShell/PowerShell#18684]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Daily build of PS Core throws exception when adding custom type that references `Newtonsoft.Json`.

```powershell
[7.3][Bukem@ZILOG] > ipmo kbsystemfunctions -Force -Verbose
VERBOSE: Loading module from path 'D:\Users\Bukem\Documents\PowerShell\Modules\kbsystemfunctions\kbsystemfunctions.psd1'.
VERBOSE: Loading 'FormatsToProcess' from path 'D:\Users\Bukem\Documents\PowerShell\Modules\kbsystemfunctions\Data\FormatData\KBConnection.Format.ps1xml'.
VERBOSE: Loading module from path 'D:\Users\Bukem\Documents\PowerShell\Modules\kbsystemfunctions\KBSystemFunctions.psm1'.
Add-Type: error CS1701: Assuming assembly reference 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' used by 'Newtonsoft.Json' matches identity 'System.Runtime, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' of 'System.Runtime', you may need to supply runtime policy
Import-Module: The module to process 'KBSystemFunctions.psm1', listed in field 'ModuleToProcess/RootModule' of module manifest 'D:\Users\Bukem\Documents\PowerShell\Modules\kbsystemfunctions\kbsystemfunctions.psd1' was not processed because no valid module was found in any module directory.
```

It's probably caused by this commit https://github.com/PowerShell/PowerShell/commit/d592ad814dc96f31b8edd2c4fb3cf89e461ae08a

**As a temporary workaround I have added `-IgnoreWarnings` switch to the `Add-Type` when compiling custom type.**

This is how it imports on PS 7.3.0 GA:
```powershell
Windows PowerShell Core v7.3.0 [Environment v3.3.0-6]

[2022-11-30 12:20:30] C:\Users\Bukem
[7.3][Bukem@ZILOG] > ipmo kbsystemfunctions -Force -Verbose
VERBOSE: Loading module from path 'D:\Users\Bukem\Documents\PowerShell\Modules\kbsystemfunctions\kbsystemfunctions.psd1'.
VERBOSE: Loading 'FormatsToProcess' from path 'D:\Users\Bukem\Documents\PowerShell\Modules\kbsystemfunctions\Data\FormatData\KBConnection.Format.ps1xml'.
VERBOSE: Loading module from path 'D:\Users\Bukem\Documents\PowerShell\Modules\kbsystemfunctions\KBSystemFunctions.psm1'.
VERBOSE: Exporting function 'Get-KBCredentialFilePath'.
VERBOSE: Exporting function 'Compare-KBObject'.
VERBOSE: Exporting function 'ConvertTo-KBNetworkMask'.
VERBOSE: Exporting function 'ConvertTo-KBNetworkPrefix'.
VERBOSE: Exporting function 'Format-KBColor'.
VERBOSE: Exporting function 'Get-KBCredentialFromFile'.
VERBOSE: Exporting function 'Get-KBNetworkID'.
VERBOSE: Exporting function 'Get-KBRandomPassword'.
VERBOSE: Exporting function 'Request-KBSpellCheck'.
VERBOSE: Exporting function 'Test-KBConnection'.
VERBOSE: Exporting function 'Save-KBCredentialToFile'.
VERBOSE: Exporting function 'Test-KBIsIPAddressInSubnet'.
VERBOSE: Exporting function 'Test-KBIsUserSessionElevated'.
VERBOSE: Importing function 'Compare-KBObject'.
VERBOSE: Importing function 'ConvertTo-KBNetworkMask'.
VERBOSE: Importing function 'ConvertTo-KBNetworkPrefix'.
VERBOSE: Importing function 'Format-KBColor'.
VERBOSE: Importing function 'Get-KBCredentialFromFile'.
VERBOSE: Importing function 'Get-KBNetworkID'.
VERBOSE: Importing function 'Get-KBRandomPassword'.
VERBOSE: Importing function 'Request-KBSpellCheck'.
VERBOSE: Importing function 'Save-KBCredentialToFile'.
VERBOSE: Importing function 'Test-KBConnection'.
VERBOSE: Importing function 'Test-KBIsIPAddressInSubnet'.
VERBOSE: Importing function 'Test-KBIsUserSessionElevated'.
[1][2022-11-30 12:20:33][00:00:00.380] C:\Users\Bukem
[7.3][Bukem@ZILOG] >
```

### Expected behavior

```console
Module should import successfully like on PS 7.3.0 GA
```

### Actual behavior

```console
Add-Type error is thrown
```

### Error details

```console
Add-Type: error CS1701: Assuming assembly reference 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' used by 'Newtonsoft.Json' matches identity 'System.Runtime, Version=7.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' of 'System.Runtime', you may need to supply runtime policy
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.0-preview.3
PSEdition                      Core
GitCommitId                    7.3.0-preview.3-499-g52ea41966297c2972313fca1ebdec537ada3eac7
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0, 5.0, 5.1, 6.0.0, 6.1.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

![image](https://user-images.githubusercontent.com/16168755/204782765-cb49d900-6642-4bd6-b0ad-80c73a2e4583.png)

**Comment by iSazonov** [comment]
Will fixed after #18656.

**Comment by briantist** [comment]
@iSazonov the PR you linked to has been closed. Any update on how/when this will be fixed? What are the affected versions of PowerShell?
